R/aes_y_lims.R

```diff
@@ -0,0 +1,615 @@
+}
+
+#' Aesthetic ticks for a given pair of numbers
+#' min_y_val - Double. Minimum y-axis value.
+#' max_y_val - Double. Maximum y-axis value.
+#' @noRd
+get_aes_ticks <- function(min_y_val, max_y_val){
+
+  if(min_y_val == max_y_val) return(NULL)
+
+  # check the argments are correctly ordered, otherwise adjust
+  if(max_y_val < min_y_val) {
+    temp_bot <- min_y_val
+    min_y_val <- max_y_val
+    max_y_val <- temp_bot
+  }
+
+  aes_band_sizes <- c(5, 10, 15, 20, 25, 30)
+
+  # differences between points by the number of ticks to include
+  aes_y_points <- list(
+    five_point = c(10, 25, 50, 75, 100),
+    six_point = c(30, 60),
+    four_point = c(20, 40, 80),
+    seven_point = c(35, 70),
+    three_point = c(15, 45, 90)
+  )
+
+  # the maximum size of the tick spacing - set to max value (100 as we adjust based on the order of magnitude)
+  max_size <- 100
+
+  # determine the difference between the two points
+  if (min_y_val <= 0 & max_y_val >= 0){
+    diff <- max_y_val + abs(min_y_val)
+
+    # both not equal to 0 - gap spans 0
+    if(min_y_val != 0 & max_y_val != 0){
+      min_val <- min(c(max_y_val, abs(min_y_val)))
+      max_val <- min(c(max_y_val, abs(min_y_val)))
+
+      max_size <- min_val
+
+      # top equal to 0 and bot below
+    } else if(min_y_val < 0){
+      max_size <- abs(min_y_val) / 2
+
+      # bot equal to 0 and top above
+    } else {
+      max_size <- max_y_val / 2
+    }
+
+  } else if (min_y_val < 0 & max_y_val < 0) {
+
+    diff <- abs(min_y_val) - abs(max_y_val)
+
+    max_size <- diff / 2
+
+  } else {
+    diff <- max_y_val - min_y_val
+
+    max_size <- diff / 2
+  }
+
+  # adjust the order of magnitude of the difference if it is less than 10 or > 100
+  order_mag <- ceiling(log10(diff))
+
+  aes_y_points <- lapply(aes_y_points, function(x) x * 10 ^ (order_mag - 2))
+  aes_band_sizes <- aes_band_sizes * 10 ^ (order_mag - 2)
+
+  order_mag_max <- ceiling(log10(max_size))
+
+  # check whether the difference fits into any of the aesthetic groupings
+  if (any(unlist(lapply(aes_y_points$five_point, dplyr::near, diff)))) {
+    band_val <- diff / 5
+
+  } else if (any(unlist(lapply(aes_y_points$six_point, dplyr::near, diff)))) {
+    band_val <- diff / 6
+
+  } else if (any(unlist(lapply(aes_y_points$four_point, dplyr::near, diff)))) {
+    band_val <- diff / 4
+
+  } else if (any(unlist(lapply(aes_y_points$seven_point, dplyr::near, diff)))) {
+    band_val <- diff / 7
+
+  } else if (any(unlist(lapply(aes_y_points$three_point, dplyr::near, diff)))) {
+    band_val <- diff / 3
+
+    # Rule 1 - If the difference is not in any of the lists, then it isn't aesthetic and we should try the max size to begin with
+  } else {
+
+    # But first check the max value is aesthetic itself - it should be most of the time
+    aes_y_points <- aes_y_points %>% unlist()
+
+    # adjust the order of magnitude if necessary
+    if(!any(unlist(lapply(aes_y_points,  dplyr::near, max_size / 10 ^ (order_mag_max - 2))))) {

```

Comment:
Why do we need the dplyr dependency here? What does `near` do that `all.equal` does not?

Reply:
haha I mean its only real benefit is that I know about near and not about all.equal...

Reply:
I'll change it then